tests/integration/security/rbac/util/rbac_util.go

```diff
@@ -56,37 +55,32 @@ func CheckRBACRequest(tc TestCase) error {
 		return fmt.Errorf("connection error with %v", err)
 	}
 
-	if tc.ExpectAllowed {
-		if !(len(resp) > 0 && resp[0].Code == connection.AllowHTTPRespCode) {
-			return fmt.Errorf("%s to %s:%d%s using %s: expected allow, actually deny",
-				req.From.Name(), req.To.Name(), req.Port, req.Path, req.Protocol)
-		}
-	} else {
+	if err != nil {
+		// If there is an error from the request, check if this request is TCP and the error contains EOF and expects to be rejected.
+		// If so, this is expected due to TCP deny request returns EOF. If one of the cases is not true, return an error.
 		if req.Port == connection.TCPPort {
-			if !strings.Contains(err.Error(), "EOF") {
+			if !(strings.Contains(err.Error(), "EOF") && tc.ExpectedRespCode == connection.DenyHTTPRespCode) {
 				return fmt.Errorf("%s to %s:%d%s using %s: expected deny with EOF error, actually %v",
 					req.From.Name(), req.To.Name(), req.Port, req.Path, req.Protocol, err)
 			}
 		} else {
-			if !(len(resp) > 0 && resp[0].Code == tc.RejectionCode) {
-				return fmt.Errorf("%s to %s:%d%s using %s: expected deny, actually allow",
-					req.From.Name(), req.To.Name(), req.Port, req.Path, req.Protocol)
-			}
+			return fmt.Errorf("test failed due to %v", err)
 		}
+	} else if !(len(resp) > 0 && resp[0].Code == tc.ExpectedRespCode) {
+		// If there is no error, simply check the response code against the expected one.
+		return fmt.Errorf("%s to %s:%d%s using %s: expected code %s, actually got %s",
+			req.From.Name(), req.To.Name(), req.Port, req.Path, req.Protocol, tc.ExpectedRespCode, resp[0].Code)
 	}
-	// Success
+
 	return nil
 }
 
 // GetRbacYamlFiles fills the template RBAC policy files with the given namespace and template files,
 // writes the files to outDir and return the list of file paths.
-func GetRbacYamlFiles(t *testing.T, outDir, namespace string, rbacTmplFiles []string) []string {
+func GetRbacYamlFiles(t *testing.T, outDir string, namespaceMapping map[string]string, rbacTmplFiles []string) []string {
```

Comment:
Also update the comment for `namespaceMapping`

Reply:
Will do. Thanks!